portfolio/src/main/java/com/google/sps/servlets/models/PublishHandler.java

```diff
@@ -0,0 +1,92 @@
+package com.google.sps.servlets.models;
+
+import com.google.api.core.ApiFuture;
+import com.google.api.core.ApiFutureCallback;
+import com.google.api.core.ApiFutures;
+import com.google.api.gax.rpc.ApiException;
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.cloud.pubsub.v1.TopicAdminClient;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.ProjectTopicName;
+import com.google.pubsub.v1.PubsubMessage;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.io.IOException;
+import java.lang.InterruptedException;
+
+public class PublishHandler {
+  private static final String PROJECT_ID = "scarletnguyen-step-2020";
+
+  private String message;
+  private ProjectTopicName topic;
+  private Publisher publisher;
+
+  public PublishHandler(String topicName) {
+    this.topic = ProjectTopicName.of(PROJECT_ID, topicName);
+    this.publisher = null;
+    this.message = null;
+  }
+
+  public void publish(String message) {
+    this.message = message;
+    try {
+      sendMessageToApi();
+    } finally {
+      if (publisher != null) terminatePublisher();
```

Comment:
4.1.1 Braces are used where optional
Braces are used with if, else, for, do and while statements, even when the body is empty or contains only a single statement.